File: linear-data-structure/LinkedLists.cpp
#include <bits/stdc++.h> 
using namespace std;

int main () {
  deque<int> dq;
  dq.push_back(4);    //inserción al final
  dq.push_back(3);
  dq.push_front(2);   //inserción al inicio
  dq.push_front(1);
  cout << dq[2] << '\n';    //acceso amortizado
  auto it = find(dq.begin(), dq.end(), 3); //búsqueda
  dq.erase(it);  //eliminación
  for(auto& e: dq) {
    cout << e << ' ';
  }
  cout << '\n';
  return 0;
}

File: linear-data-structure/LinkedLists.java
import java.util.*;
public class LinkedLists {
  public static void main (String args[]) {
    var lk = new LinkedList<Integer>();
    lk.addLast(4);    //inserción al final
    lk.addLast(3);                          
    lk.addFirst(2);   //inserción al inicio
    lk.addFirst(1);
    System.out.println(lk.get(2)); //acceso lineal
    int idx = lk.indexOf(3); //búsqueda
    lk.remove(idx); //eliminación
    for(var e: lk) {
      System.out.print(e + " ");
    }
    System.out.println();
  }
}
